questionnaire/questionnaire/questionnaire.js

```diff
@@ -190,6 +190,40 @@ function createQuestionnaire({
         return resolvedVars;
     }
 
+    function getDependentQuestionIds(questionId, sortingInstructions) {
+        const questionOrdering = sortingInstructions[questionId];
+        return questionOrdering || [];
+    }
+
+    function dependentsInIncorrectOrder(values, valueIndex, sortingInstructions) {
+        if (valueIndex === 0) {
+            return false;
+        }
+        const currentQuestionId = values[valueIndex].id;
```

Comment:
Use Destructuring: Utilize destructuring where applicable to simplify code and improve readability.
const { id: currentQuestionId } = values[valueIndex];
const { id: previousQuestionId } = values[valueIndex - 1];

Reply:
I think that destructuring is actually less readable here, as we are not assigning the Id property to a variable of the same name

Reply:
{ id: currentQuestionId }: This is an object destructuring syntax. It means that from the object retrieved from values[valueIndex], we want to extract the property id and assign it to a variable named currentQuestionId.
Happy to leave as is though as arguably this is less readable